File: src/main/java/com/dontwaste/model/entity/product/Product.java
package com.dontwaste.model.entity.product;

import com.dontwaste.model.base.BaseDate;
import com.dontwaste.model.entity.ProductTemplate;
import com.dontwaste.model.entity.Branch;
import lombok.*;

import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
@Builder
@Entity
@Table(name = "PRODUCT")
public class Product extends BaseDate {

    @Column(name = "STARTING_PRICE", nullable = false)
    private Double startingPrice;

    @Column(name = "PRICE", nullable = false)
    private Double price;

    @Column(name = "QUANTITY")
    private Integer quantity;

    @OneToOne
    @JoinColumn(name = "BRANCH_ID", referencedColumnName = "ID")
    private Branch branch;

    @OneToOne
    @JoinColumn(name = "TEMPLATE_ID", referencedColumnName = "ID")
    private ProductTemplate productTemplate;


}

File: Front/src/app/-Models-/product.ts
import {Branch} from "./branch";
import {ProductTemplate} from "./product-template";

export class Product {
  id: number;
  startingPrice: number;
  price: number;
  quantity: number;
  totalPrice: number;
  branch: Branch;
  productTemplate: ProductTemplate;
  product: Product;

  constructor(id: number,
              startingPrice: number,
              price: number,
              quantity: number,
              branch: Branch,
              productTemplate: ProductTemplate,
              totalPrice: number,
              product: Product) {
    this.id = id;
    this.startingPrice = startingPrice;
    this.price = price;
    this.quantity = quantity;
    this.branch = branch;
    this.productTemplate = productTemplate;
    this.product = product;
    this.totalPrice = totalPrice * quantity;
  }
}
